xbmc/threads/Event.cpp

```diff
@@ -115,11 +105,9 @@ namespace XbmcThreads
     // signaled and sets 'signaled' to the first one it
     // finds.
     // ==================================================
-    signaled = NULL;
-    for (std::vector<CEvent*>::iterator iter = events.begin();
-         signaled == NULL && iter != events.end(); ++iter)
+    signaled = nullptr;
+    for (auto* cur : events)
```

Comment:
Yep, `*` is not required here - it's actually on purpose. I wanted it to be immediately visible that you're iterating over plain pointers.